radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/ForkManager.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * (C) Copyright 2021 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ */
+
+package com.radixdlt.statecomputer.forks;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.hash.HashCode;
+import com.radixdlt.statecomputer.LedgerAndBFTProof;
+import com.radixdlt.store.EngineStore;
+import com.radixdlt.sync.CommittedReader;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Manages forks and their transitions. There are two kinds of forks:
+ * - a list forks executed at fixed epochs
+ * - an optional candidate fork that is switched on based on a predicate (in most cases stake voting)
+ *
+ * All forks must be executed in order, and a candidate fork can only be considered
+ * when we're already running the latest fixed epoch fork.
+ */
+public final class ForkManager {
```

Comment:
Name is kind of confusing. Manager implies that it is managing state when in fact it is just holding immutable configuration.

Reply:
I like the old `Forks` name.